src/datasource.ts

```diff
@@ -0,0 +1,345 @@
+        return compartments.map(n => {
+          return { text: n.name, value: n.ocid };
+        }); 
+      }   
+    }    
+    return [];
+  }
+
+
+  getJsonData() {
+    return this.jsonData;
+  }
+  
+  getVariables() {
+    const templateSrv = getTemplateSrv();
+    return templateSrv.getVariables().map((v) => `$${v.name}`);
+  }
+
+  getVariablesRaw() {
+    const templateSrv = getTemplateSrv();
+    return templateSrv.getVariables();
+  }  
+
+
+ // **************************** Template variables helpers ****************************
+
+  /**
+   * List all variable names optionally filtered by regex or/and type
+   * Returns list of names with '$' at the beginning. Example: ['$dimensionKey', '$dimensionValue']
+   *
+   * Updates:
+   * Notes on implementation :
+   * If a custom or constant is in  variables and  includeCustom, default is false.
+   * Hence,the varDescriptors list is filtered for a unique set of var names
+   */
+
+  /**
+   * @param varName valid varName contains '$'. Example: '$dimensionKey'
+   * Returns true if variable with the given name is found
+   */
+  isVariable(varName: string) {
+    const varNames = this.getVariables() || [];
+    return !!varNames.find((item) => item === varName);
+  }
+
+
+  // main caller to call resource handler for get call
+  async getResource(path: string): Promise<any> {
+    return super.getResource(path);
+  }
+  // main caller to call resource handler for post call
+  async postResource(path: string, body: any): Promise<any> {
+    return super.postResource(path, body);
+  }
+
+
+  async getTenancies(): Promise<OCIResourceItem[]> {
+    return this.getResource(OCIResourceCall.Tenancies).then((response) => {
+      return new ResponseParser().parseTenancies(response);
+    });
+  }
+
+  async getSubscribedRegions(tenancy: string): Promise<string[]> {
+    if (this.isVariable(tenancy)) {
+      let { tenancy: var_tenancy} = this.interpolateProps({tenancy});
+      if (var_tenancy !== "") { 
+        tenancy = var_tenancy
+      }      
+    }
+    if (tenancy === '') {
+      return [];
+    }
+    const reqBody: JSON = {
+      tenancy: tenancy,
+    } as unknown as JSON;
+    return this.postResource(OCIResourceCall.Regions, reqBody).then((response) => {
+      return new ResponseParser().parseRegions(response);
+    });
+  }
+
+  async getCompartments(tenancy: string): Promise<OCIResourceItem[]> {
+    if (this.isVariable(tenancy)) {
+      let { tenancy: var_tenancy} = this.interpolateProps({tenancy});
+      if (var_tenancy !== "") { 
+        tenancy = var_tenancy
+      }      
+    }   
+    if (tenancy === '') {
+      return [];
+    }
+    const reqBody: JSON = {
+      tenancy: tenancy,
+    } as unknown as JSON;
+    return this.postResource(OCIResourceCall.Compartments, reqBody).then((response) => {
+      return new ResponseParser().parseCompartments(response);
+    });
+  }
+
+  async getNamespacesWithMetricNames(
```

Comment:
I can still see these functions which are not required in logs plugin

Reply:
Done!